### PR TITLE
BI-1616

### PR DESCRIPTION
**Pull request by nickpalladino**
# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1616?atlOrigin=eyJpIjoiMDY3MzAwZjNlNTM4NDQ4MzkwMmJhZDUzN2UxYjdmZWMiLCJwIjoiaiJ9

- Added support for filtering on brapi lists endpoint

# Dependencies
- bi-web feature/BI-1616

# Testing
- Test filtering, sorting and pagination in Germplasm Lists table to verify it's working as expected.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [x] I have tested that my code works with both the brapi-java-server and BreedBase
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<please include a link to TAF run>_

### CODE REVIEW

File: src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java [hunk 1]
```diff
@@ -90,10 +92,24 @@ public BrAPIServerInfoResponse serverinfo() {
...
     public HttpResponse getLists(@PathVariable("programId") UUID programId, HttpRequest<String> request,
-                                 @QueryValue @QueryValid(using = ListQueryMapper.class) @Valid BrapiQuery queryParams
+                                 @QueryValue @QueryValid(using = ListQueryMapper.class) @Valid ListQuery queryParams
     ) throws DoesNotExistException, ApiException {
-        List<BrAPIListSummary> brapiLists = germplasmService.getGermplasmListsByProgramId(programId, request);
-        return ResponseUtils.getBrapiQueryResponse(brapiLists, listQueryMapper, queryParams);
+        List<BrAPIListSummary> brapiLists;
+
+        if (queryParams.getListType() == null) {
+            // TODO: in future return all list types but for now just return germplasm
```
Comment (timparsons) [inline src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java:100]: Why not just have this method return a 400 if the listType isn't specified?

File: src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java [hunk 2]
```diff
@@ -90,10 +92,24 @@ public BrAPIServerInfoResponse serverinfo() {
...
+            brapiLists = germplasmService.getGermplasmListsByProgramId(programId, request);
+        } else {
+            // TODO: return appropriate lists by type, only germplasm currently
+            switch(queryParams.getListType()) {
+                case "germplasm":
+                default:
+                    brapiLists = germplasmService.getGermplasmListsByProgramId(programId, request);
```
Comment (timparsons) [inline src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java:107]: Similar to above, maybe a 400 should be returned here.

**Review by timparsons — CHANGES_REQUESTED** [review]
Left some comments about how to handle the listType query param

File: src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java (hunk 1 above)
Comment (timparsons) [reply src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java:100]: Ah...good point.  Looked back at the spec, and you're spot on.  I'll approve this!

**Review by timparsons — APPROVED** [review]

File: src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java (hunk 1 above)
Comment (nickpalladino) [reply src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java:100]: cool thanks!

**Review by dmeidlin — APPROVED** [review]

**Pull request merged**